src/hostMeeting/hostMeeting.controller.js

```diff
@@ -81,6 +85,69 @@
                 return !!user.nextMovie;
             });
         }
+
+        function saveMeeting() {
+            var presentUsers = _.reduce(vm.presentUsers, function (presentUsers, item) {
+                presentUsers[item.id] = true;
+                return presentUsers;
+            }, {});
+
+            meetingApi.saveMeeting(new Date(), presentUsers, currentMovie.name, currentMovieUser.userId);
+        }
+
+        function selectWinner() {
+            var deferred = $q.defer();
+            meetingApi.getAll().$loaded().then(function (meetings) {
+                var chances = {};
+                getPossibleWinners(chances);
+                generateProbabilitiesFromMeetings(chances, meetings);
+                var winner = selectRandomWinner(chances);
+                deferred.resolve(winner);
+            });
+            return deferred.promise;
```

Comment:
Using `$q.defer()` is rarely needed, and is a very common promise anti-pattern. Try this as an alternative:

```javascript
function selectWinner() {
   return meetingApi.getAll().$loaded().then(function (meetings) {
      var chances = {};
      getPossibleWinners(chances);
      generateProbabilitiesFromMeetings(chances, meetings);
      return selectRandomWinner(chances);
   });
}
```